labs/b4.md

```diff
@@ -0,0 +1,269 @@
+layout: lab
+---
+
+## What is a distribution?
+
+A distribution is a combination of the a kernel, init system, libraries,
+drivers, and other critical software. There are many distributions, for Linux
+and other kernels as well, such as BSD and Solaris.
+
+## What should I use then?
+
+Currently, there are around 300 actively developed Linux distributions listed
+on [DistroWatch](https://distrowatch.com). It might be a bit overwhelming with
+so many choices. Popular distributions include Debian, Ubuntu, Arch Linux,
+CentOS, and Red Hat Enterprise Linux, but in this course, we’ll only be using
+Debian.
+
+### What is Debian?
+
+Debian is a distribution developed by the [Debian Project](https://debian.org)
+starting in 1993. This distribution strives to provide a completely free
+(libre) software experience. In the context of free software, "libre" means
+that the source code is available under a permissive license, enabling
+downstream modification and rediistribution. Many other distributions are based
+on Debian, as Debian releases are usually very stable, meaning that the
+software provided by the distribution has been tested to be as bug-free as
+possible. One such distribution is Ubuntu, which is very popular and has
+spawned its own family of distributions. What defines the Debian family of
+distributions is the use of the DEB package format and the dpkg package
+manager.
+
+## But what about these “package managers”?
+
+If you come from the Windows or MacOS ecosystems, you might be wondering what a
+package manager is. On those operating systems, the most common method of
+installing new programs is to run an installer executable that unpacks and
+copies files into the correct location, and updates the OS registry to reflect
+this.
+
+Enter the package manager, a centralized way to install, update, and remove
+software from your computer using verified sources called repositories.  Most
+package managers work in the same way: the package manager gets a list of
+packages from the repository, and then, when asked to install a package,
+fetches the package from the repository, verifies that it is legit, and
+installs it. A package manager is like a librarian. When a patron wants to read
+a book, the librarian consults a catalog (which is the package database) and
+then fetches the book from the shelf and gives it to the patron (installing the
+package).  Sometimes, the librarian has to update the catalog because new
+editions of the books were added (updating the package database). When the
+patron is done, the librarian knows where to return the book (cleaning up after
+removing a package).  Contrast this to the approach on Windows, where dedicated
+uninstaller programs are occasionally necessary as the OS does not know where
+programs install their own files.
+
+## Debian: An Example
+
+In this class, we will be focused on using Debian. As noted before, Debian uses
+`apt` or `dpkg` as its package manager.
```

Comment:
Explain the difference between `apt` and `dpkg` (and probably tell them that they should mostly be using `apt`)